ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activityTextAnswer.component.ts

```diff
@@ -1,65 +1,66 @@
 import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChange } from '@angular/core';
+import { Observable, Subject, of } from 'rxjs';
+import { debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+
 import { ActivityTextQuestionBlock } from '../../../models/activity/activityTextQuestionBlock';
 import { InputType } from '../../../models/activity/inputType';
 import { TextSuggestion } from '../../../models/activity/textSuggestion';
 import { SuggestionMatch } from '../../../models/suggestionMatch';
 import { TextSuggestionProvider } from '../../../models/activity/textSuggestionProvider';
-import { Observable, Subject, of } from 'rxjs';
-import { debounceTime, switchMap } from 'rxjs/operators';
 
 const MIN_INPUT_LENGTH_FOR_TYPEAHEAD = 3;
 
 @Component({
     selector: 'ddp-activity-text-answer',
     template: `
-    <ddp-question-prompt [block]="block"></ddp-question-prompt>
-    <ng-container *ngIf="isTextInputType(block)">
-        <mat-form-field  class="example-input-field" [floatLabel]="block.label ? 'always' : null">
-            <mat-label *ngIf="block.label" [innerHTML]="block.label"></mat-label>
-            <input matInput [(ngModel)]="block.answer"
-                [minlength]="block.minLength"
-                [maxlength]="block.maxLength"
-                [pattern]="block.regexPattern"
-                [placeholder]="placeholder || block.placeholder"
-                [disabled]="readonly"
-                [attr.data-ddp-test]="'answer:' + block.stableId"
-                (input)="inputSubject.next(textInput.value)"
-                (change)="onChange(textInput.value)"
-                [matAutocomplete]="autoCompleteFromSource"
-                #textInput="ngModel">
-            <mat-autocomplete #autoCompleteFromSource="matAutocomplete"
-                (optionSelected)="onChange(textInput.value)" class="autoCompletePanel">
+        <ddp-question-prompt [block]="block"></ddp-question-prompt>
+        <ng-container *ngIf="isTextInputType(block)">
+            <mat-form-field class="example-input-field" [floatLabel]="block.label ? 'always' : null">
+                <mat-label *ngIf="block.label" [innerHTML]="block.label"></mat-label>
+                <input matInput [(ngModel)]="block.answer"
+                       [minlength]="block.minLength"
+                       [maxlength]="block.maxLength"
+                       [pattern]="block.regexPattern"
+                       [placeholder]="placeholder || block.placeholder"
+                       [disabled]="readonly"
+                       [attr.data-ddp-test]="'answer:' + block.stableId"
+                       (input)="inputSubject.next(textInput.value)"
+                       (ngModelChange)="onChange($event)"
```

Comment:
It occurs to me that there might be a simpler solution: looks like root cause of the problem is that we originally capturing text using the "onchange" event and this event does not trigger on initial input.
So probably simpler to just tweak existing code to do same thing it is doing now, but go from:
```typescript
(change)="onChange(textInput.value)
```
to
```typescript
(input)="onChange(textInput.value)
```
(You could also consider renaming onChange method to match the name of the event that triggers call)

And since you still want to execute an `inputSubject.next(textInput.value)` , you can just put this line in the onChange method.

Approach also has virtue of avoiding adding more instance variables (every time we add an instance variable, particularly in components, my anxiety goes up. More state to worry about).

By the way: this is a suggestion. Please evaluate and use if you think makes sense to use.

Reply:
Also, we should consider applying fix to the other textAnswer cases included in this component (like the `isEssayInputType()` case). Not seeing it now because we are not including essay questions in dialogs, but this is a bug waiting to happen.

Reply:
Agree, the first implementation was overcomplicated. Thank you for the tip! 

Refactored at 9be26f0d422939a3407d3a6e149be665068dd7cc :
- kept `onChange` for email component as to emit changed value immediately
- used `(input) = "inputSubject.next(textInput.value)"` for both TextInput and EssayInput  in order to fire the changed textInput value. 
  Added a debounce there to decrease number of emitted values (patch answer requests).
  The `inputSubject` is also used for getting filteredSuggestions$ at the same time with the same `input event`    
  (because `Rxjs Subject` allows to share a single execution with multiple observers).